cpp/src/io/parquet/writer_impl.cu

```diff
@@ -85,6 +87,49 @@ parquet::Compression to_parquet_compression(compression_type compression)
   }
 }
 
+size_type column_size(column_view const& column, rmm::cuda_stream_view stream)
+{
+  if (column.num_children() == 0) {
+    return is_fixed_width(column.type()) ? size_of(column.type()) * column.size() : 0;
+  }
+
+  if (column.type().id() == type_id::STRING) {
+    auto const scol = strings_column_view(column);
+    return cudf::detail::get_value<size_type>(scol.offsets(), column.size(), stream) -
+           cudf::detail::get_value<size_type>(scol.offsets(), 0, stream);
+  } else if (column.type().id() == type_id::STRUCT) {
+    auto const scol = structs_column_view(column);
+    size_type ret   = 0;
+    for (int i = 0; i < scol.num_children(); i++) {
+      ret += column_size(scol.get_sliced_child(i), stream);
+    }
+    return ret;
+  } else if (column.type().id() == type_id::LIST) {
+    auto const lcol = lists_column_view(column);
+    return column_size(lcol.get_sliced_child(stream), stream);
+  }
+
+  return 0;
+}
+
+// checks to see if the given column has a fixed size.  This doesn't
+// check every row, so assumes string and list columns are not fixed, even
+// if each row is the same width.
+// TODO: update this if FIXED_LEN_BYTE_ARRAY is ever supported for writes.
+bool is_col_fixed_width(column_view const& column)
+{
+  if (is_fixed_width(column.type())) {
+    return true;
+  } else if (column.type().id() == type_id::STRUCT) {
+    for (int i = 0; i < column.num_children(); i++) {
+      if (not is_fixed_width(column.child(i).type())) { return false; }
+    }
+    return true;
+  }
+
+  return false;
```

Comment:
Should we recursively call `is_col_fixed_width? My (simpler?) version:
```suggestion
  if (column.type().id() == type_id::STRUCT) {
    return std::all_of(column.child_begin(), column.child_end(), is_col_fixed_width);
  }

  return is_fixed_width(column.type());
```

Reply:
Sounds good. Like above, still not sure why there's a struct as a leaf.